File: src/main/java/com/nridwan/model/PackageList.java
package com.nridwan.model;

import java.util.HashMap;

public class PackageList {
    public HashMap<String, String> getPackages() {
        return packages;
    }

    public void setPackages(HashMap<String, String> packages) {
        this.packages = packages;
    }

    public void merge(PackageList others) {
        others.packages.forEach((k, v) -> {
            if(v.contains("#") || !packages.containsKey(k) || !packages.get(k).contains("#")) {
                packages.put(k, v);
                return;
            }
            packages.put(k, v+"#"+packages.get(k).split("#")[1]);
        });
    }

    private HashMap<String, String> packages = new HashMap<>();

    @Override
    public String toString() {
        return "PackageList{" +
                "packages=" + packages +
                '}';
    }
}

File: standalone_src/subpm/buildSrc/src/main/kotlin/com/subpm/model/PackageList.kt
package com.subpm.model

import java.util.HashMap

class PackageList {
    fun merge(others: PackageList) {
        others.packages.forEach { k: String, v: String? ->
            if (v?.contains("#") === true || !packages.containsKey(k) || packages.get(k)?.contains("#") !== true) {
                packages.put(k, v!!)
                return@forEach
            }
            packages.put(k, v + "#" + packages.get(k)!!.split("#".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray().get(1))
        }
    }

    var packages = HashMap<String, String>()
    override fun toString(): String {
        return "PackageList{" +
                "packages=" + packages +
                '}'
    }
}